### PR TITLE
you may now hang yourself (and others)

### PR DESCRIPTION
**Pull request by mc-oofert**
## About The Pull Request
ports nooses from beestation, with basically a 80% code overhaul
you get it by crafting a noose with 30 cable coil if youre standing directly over a chair

https://github.com/tgstation/tgstation/assets/70376633/ee0419d5-5413-49ef-b7a3-a7a34bf80d54

## Why It's Good For The Game

its a pretty funny option to kill yourself like imagine walking into the theatre backroom and you see the clown hanging from the goddamn ceiling
also you can hang heads or revs or traitors to send a message

## Changelog
:cl:
add: nooses, craftable with 30 cable
/:cl:

### CODE REVIEW

File: code/game/objects/structures/noose.dm [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+				if(!HAS_TRAIT(buckled_mob, TRAIT_NOBREATH))
+					buckled_mob.adjustOxyLoss(5)
+					if(prob(30))
+						buckled_mob.emote("gasp")
+				if(prob(20))
+					var/flavor_text = list(span_suicide("[buckled_mob]'s legs flail for anything to stand on."),\
+											span_suicide("[buckled_mob]'s hands are desperately clutching the noose."),\
+											span_suicide("[buckled_mob]'s limbs sway back and forth with diminishing strength."))
+					buckled_mob.visible_message(pick(flavor_text))
+			playsound(buckled_mob.loc, 'sound/effects/noose_idle.ogg', 25, 1, -3)
```
Comment (SmArtKar) [inline code/game/objects/structures/noose.dm:131]: You could use early returns here, and probably in a couple of other places.

**Comment by Exester509** [comment]
Can you hang others?

**Comment by mc-oofert** [comment]
> Can you hang others?

yeah

**Comment by mc-oofert** [comment]
> This causes incredible amounts of lag when people build multiple in the halls and produce 10+ monkeys to hang in them. At least, that's what I remember. Is the lag still a problem and if so how many hanged monkeys before it kicks in.

image
none

**Comment by Wallemations** [comment]
Maybe if this was a result of a `suicide()` call but I'm not sure how I feel about this otherwise
I know a lot of stuff in this game is graphic but this just feels a bit much
Idk I could be overreacting

**Comment by dendyd00m** [comment]
it also makes me a little uncomfortable. it crosses just enough from "cartoony violence" to "this is a bit too real for me"

things like guns and shooting yourself are all content that are fully visible and expected going into the game. the guillotine was a good compromise in my mind for a flashy and over the top execution method.

as an admin i've also had to deal with countless cases of players telling other players to hang themselves when they get mad OOCly, and it was easy to deal with because there was no in-game equivalent. this offers plausible deniability that makes those issues not so simple to deal with.

just in general, while i appreciate that this is a well-intentioned mechanic that wants to bring more interaction to the game, it doesn't feel great to see my cute spaceman sprite swinging from a rope and choking to death.

**Review by MrMelbert — CHANGES_REQUESTED** [review]
1000% don't want this as a craft-able thing you can put other people in, i think it will only be used in cringe OOC larp moments

i think at most it should be a cable coil suicide-act

**Comment by necromanceranne** [comment]
https://github.com/tgstation/tgstation/pull/23654 
https://github.com/BeeStation/BeeStation-Hornet/pull/1438

The putrid ghost of hippie continues to haunt us all

**Comment by Iamgoofball** [comment]
this is a gross PR that we don't need

**Comment by Moltovz** [comment]
> this is a gross PR that we don't need

you said the same thing in 2017 and I agree still

**Comment by Exester509** [comment]
How is this more graphic than a hallway lined with cut heads on spikes

**Comment by Iamgoofball** [comment]
not interested in arguing about this one, just close the PR

**Comment by Jacquerel** [comment]
SS13 is a game of dark humour which is often marcarbre and unpleasant, we are perfectly happy to have people blow off their own heads with russian roulette and characters regularly end rounds in states that are entirely mutilated. The setting and codebase of TG even does feature prejudice based on species as a core part of the worldbuilding.

Given all that it's not clear or easy for me to elaborate on what makes _this_ cross a line that other things we put in the game do not, but something about it does not pass the sniff test.
I'm putting this on DNM instead of closing this in case some other maintainer disagrees and can make a really good case for why it's fine, but personally I think the gameplay dynamics introduced by being able to lynch other players by hanging would not be positive, and even the sight of going into a room to see a corpse dangling from a noose via suicide would be a level of disturbing for people that's above what we usually expect in the game.

**Comment by DATA-xPUNGED** [comment]
The way the guy on the noose moves around is disturbing, Not a fan of it.

**Comment by DaCoolBoss** [comment]
I'm not feeling the tone argument here. If it's tame enough for Looney Toons and the Addams family it's tame enough for this I feel. For what it's worth I've played on servers with the nooses and have never seen them used for enacting a lynching but have seen it used in court RP, on headrevs during a rev round, and with legion skeletons in them for a haunted house.

**Comment by carlarctg** [comment]
image

I agree with this statement. This PR in its current form is a little bit too edgy and not quite funny enough to be a good and thematic addition.

I propose instead adding the ability to hang corpses/people upside down from their feet up, kinda like the Predator's victims. It's not nearly as 'graphic', still lets you display corpses, while adding some rather silly potential such as someone flailing to get free.

**Comment by necromanceranne** [comment]
I was literally thinking the exact same thing actually. Upside down from the feet is a little more broadly interesting and a neat way to capture people.

**Comment by moocowswag** [comment]
Its literally exactly as i imagined it get out of my head.

**Comment by LePiromano** [comment]
i understand it's disturbing but we already have worse shit, sure ss13 is more comedy than anything and most of the fucked up shit you can do in the game is actually kinda funny while this is just flat out macabre, but i personally don't see any issue with this that isn't already present in the game in shit like heads on pikes, or strangling someone with your bare hands, or slitting their throat and shit

**Comment by DaCoolBoss** [comment]
> would this type of hanging be more suitable https://www.youtube.com/watch?v=Uw8flMqLBOY
> 
> Speaking of ss13 largely having cartoon violance are there literally any cartoon moments at all that involve hanging? i cant think of a looney toons moment? maybe this decision is still consistent with the genre.
> 
> The only example i can think of isnt a cartoon its a live action disney movie, the opening to the pirates of the Caribbean.

It's a staple in westerns and horror movies. Like I said, it's happened a few times in Looney Toons, and it's been used as a gag in the Simpsons a few times.

**Comment by kawaiinick** [comment]
same game that features real life hard drugs

**Comment by itseasytosee** [comment]
There is obviously a tonal element to it. I really think it would be possible to push some more sci-fi execution/suicide methods. Execution by spacing is probably similarly brutal but its permissible by the fact that it contributes the the setting in some way, and doesn't have the same baggage that hanging does. 
People associate the noose with some pretty depressing subject matter, and thematic are a big part of this discussion.

Any "But we have X comparable thing" arguments are dumb. At the end of the day, something like this is a matter of themeatics and vibes.
If this was a western setting I think you could get away with it, but it has different connotations in space.

**Comment by Loafin34** [comment]
we have meatspike

**Comment by kawaiinick** [comment]
> If this was a western setting I think you could get away with it, but it has different connotations in space.

the game is set in the space frontier though, so it has elements of western

**Comment by MTandi** [comment]
18+ video game with hundreds of ways to kill yourself violently:

> uncomfortable, gross

Meanwhile cartoons for kids from Disney:

image
image

**Comment by MTandi** [comment]
I don't see any reason why people would think about this differently compared to other suicide methods we have.

On the other hand I imagine scenes where people hang robots or individuals with an air pump implant hilarious. And it would be a shame if this won't be added because someones is especially sensitive to this type of suicide for some reason.

**Comment by LSommers** [comment]
Anyone who's a fat fuck should snap it instantly, and clowns should completely botch any attempt to use it unless they explicitly do a *suicide action. Just make it need a gallows platform or something to make it less easily created and thus spammable or something idk.

There's nothing more inherently grim/"eww" about this vs Sec lining people up against a wall, tiders/ashwalkers putting heads on spears, people getting tied to the stake and burned alive by the chaplain, atmosians making a straight-up fucking gas chamber, etc etc etc.

All of these things are kinda fucked but generally result in some cartoon shit when used by the players (IE atmosians making a BZ chamber and putting a ling in Super GBJ, the Chaplain putting a fireproof guy on the stake to prove witchcraft, etc.), what exactly makes this different?

**Comment by MTandi** [comment]
The noose created by a clown should break down back to 30 cables because of TRAIT_CLUMSY. So when a clown wants to hang himself, he should ask other people to make a noose for him.

**Comment by Cheshify** [comment]
Let's fucking not

**Comment by Vect0r2** [comment]
image
Insta-close thumbs up own PR

**Comment by DaCoolBoss** [comment]
Lower its integrity a bunch from the default of 300, so people can destroy it easily with knives or by shooting it. Shooting it especially.

**Comment by optimumtact** [comment]
not reading all that but this came from beestation and that's reason enough for me to deny it.

**Comment by SomeoneYouProbablyKnow** [comment]
> If it's tame enough for Looney Toons and the Addams family it's tame enough for this I feel. For what it's worth I've played on servers with

It's not a strict line like **thing A** allowed **thing B** not allowed. It's not just that it's a noose. It's the framing, the chair, the swaying. It's voyeuristic. 

> Execution by spacing is probably similarly brutal but its permissible by the fact that it contributes the the setting in some way, and doesn't have the same baggage that hanging does.

Real people don't have relatives who jumped into hard vaccum or got cut in half with an energy sword. The most "real" suicide option we have are guns probably and it doesn't have its own little animation nor does it leave a special reminder in the environment screaming "SUICIDE BY *THING*"

And on the topic of heads on spikes, it's probably the most gruesome thing we have in the game. An outlier. And honestly kinda yuck, but it IS in the game already, doesn't mean we have to push the line even further.

> And it would be a shame if this won't be added because someone is especially sensitive to this type of suicide for some reason.

No it really wouldn't be a shame. I'd say it would be a good outcome if not merging this PR prevented that kind of distress for people. 

It doesn't add enough for thr actual game to sacrifice stuff for it and it would set a bad precedent.

**Comment by itseasytosee** [comment]
> > Execution by spacing is probably similarly brutal but its permissible by the fact that it contributes the the setting in some way, and doesn't have the same baggage that hanging does.
> 
> Real people don't have relatives who jumped into hard vaccum or got cut in half with an energy sword. The most "real" suicide option we have are guns probably and it doesn't have its own little animation nor does it leave a special reminder in the environment screaming "SUICIDE BY _THING_"
> 

I mostly agree with you yeah. I don't think it really fits our theming and I think it will harsh the setting based on how I expect to see it used in game.

**Comment by LSommers** [comment]
> The noose created by a clown should break down back to 30 cables because of TRAIT_CLUMSY. So when a clown wants to hang himself, he should ask other people to make a noose for him.

Balloon noose. Furious rubber squeaking nose ala balloon animal creation, inventory fiddling noise, honk, pause, sad trombone. ":s WHO LEFT THIS SHITASS CLOWN WITH THEIR BALLOONS IN THE CELL? NOW WE CAN'T EVEN GULAG HIM"

**Comment by ArcaneMusic** [comment]
I'll leave my two cents here for posterity: I totally agree that this one just feels too uncomfortable to add to the game.

Most suicide_act() code in the game is gruesome, gory, violent, and somewhat crass. That's a given. But it also stands to reason that it's usually somewhat theatrical or goofy, because at the core we all understand that it's a player deciding to stop playing the round, and wanting to show that to other players. By either having a player hang themself or someone else, it goes into "realworld somber & distressing" far deeper than most other examples. Also, the fact another player can do it to you just feels a little sick, even if you DIDN'T use a suicide verb.

Last note: The Ceiling does not exist. I hope someone got fired for THAT blunder!

**Comment by ILLUZORR** [comment]
merge

**Pull request closed without merging**